### DISCUSSION TITLE
Address Mapping

### DISCUSSION THREAD

**Discussion by pedrovalido** [Fujicracy/fuji-v2#179] [General]
From the discord thread:
Hey, this is a thread to document the consensus/standard of the AddressMapper and respective mappings. I was talking to Daigaro and accidentally did an overlapping mapping with the compound tokens to use with morpho and the ones that were already defined for Compound (Morpho uses compound ctokens) .

We’re trying to figure out which of the following ways is best:
1) Have the same mapping provider1, tokenA, tokenB and provider2, tokenA, tokenB but have a clear understanding of where the mappings are being used
2) have no duplicates in the mappings resulting in MorphoCompound using the same mapping as compound to fetch compound_V2, tokenA, cToken

As discussed, we've decided to go with option 2 for now.
Let's use this discussion to further document this.

Deployed Address Mappers for testing purposes:
https://arbiscan.io/address/0x9b66e949277d6b5de1e1099242c57cdaa53782b5#code
https://optimistic.etherscan.io/address/0x4cb46032e2790d8ca10be6d0001e8c6362a76ada
https://polygonscan.com/address/0xe7aa20127f910dc20492b320f1c0cab12dfd4153
https://etherscan.io/address/0x529ee84bfe4f37132f5f9599d4cc4ff16ee6d0d2#code
https://gnosisscan.io/address/0x2b3a214b2218f1ce35e94ee3ba14d8af1712ab04

**Note: we are always mapping the weth and wbtc instead of the eth and btc tokens because our vaults use wrapped token instead of the native ones. The interaction with a protocol that uses the native token, is handled by the code (wrapping/unwrapping).**

# Providers and their mappings

## arbitrum

AaveV3 (not using address mapper)

### **DForce (providerName = “DForce”)**
```
iUSDC(0x8dc3312c68125a94916d62B97bb5D925f84d4aE0)
iETH(0xEe338313f022caee84034253174FA562495dcC15)
iDAI(0xf6995955e4B0E5b287693c221f456951D612b628)
iWBTC(0xD3204E4189BEcD9cD957046A8e4A643437eE0aCC)
```

### **Hundred (providerName = “Hundred”)**
```
hETH(0x8e15a22853A0A60a0FBB0d875055A8E66cff0235)
hUSDC(0xfcd8570ad81e6c77b8d252bebeba62ed980bd64d)
hWBTC(0xb4300e088a3ae4e624ee5c71bc1822f68bb5f2bc)
hDAI(0x6bb6ebCf3aC808E26545d59EA60F27A202cE8586)
```

Radiant (not using address mapper)

### **WePiggy (providerName = “WePiggy”)**
```
pETH(0x17933112E9780aBd0F27f2B7d9ddA9E840D43159)
pWBTC(0x3393cD223f59F32CC0cC845DE938472595cA48a1)
pUSDC(0x2Bf852e22C92Fd790f4AE54A76536c8C4217786b)
pDAI(0xDe39Adfb2025D2aA51f6fD967e7C1753215f1905)
```

## gnosis

Agave (not using address mapper)

### **Hundred (providerName = “Hundred”)**
```
hETH(0x626915a1beFf83240ea1290a3269d4849a7b599D)
hxDAI(0x6eDCB931168C9F7C20144f201537c0243b19dCA4)
hWBTC(0x86b7966C756e5D8827594D59b52D9a0D7B1AA3dC)
hUSDC(0xbd193db8a909cAC57Cdb981Ea81B5dc270287F19)
```

## mainnet

AaveV2 (not using address mapper)

AaveV3 (not using address mapper)

### **CompoundV2 (providerName = “Compound_V2”)**
```
cDAI(0x5d3a536E4D6DbD6114cc1Ead35777bAB948E3643)
cETH(0x4Ddc2D193948926D02f9B1fE9e1daa0718270ED5)
cUSDC(0x39AA39c021dfbaE8faC545936693aC917d5E7563)
cWBTC(0xC11b1268C1A384e55C48c2391d8d480264A3A7F4)
```

### **CompoundV3 (providerName = “Compound_V3”)**
Address Mapper has to be set with nested mapping - (collateralAsset, debtAsset) -> cMarketV3. 
cMarketv3 is the market of the debtAsset (if the goal operation is to use a BorrowingVault). 
When using a YieldVault, the collateralAsset is the asset to be deposited and debtAsset is address(0).
```
cUSDCv3(0xc3d688B66703497DAA19211EEdff47f25384cdc3)
cWETHv3(0xA17581A9E3356d9A858b789D68B4d866e593aE94)
```

### **DForce (providerName = “DForce”)**
```
iWBTC(0x5812fCF91adc502a765E5707eBB3F36a07f63c02)
iETH(0x5ACD75f21659a59fFaB9AEBAf350351a8bfaAbc0)
iUSDC(0x2f956b2f801c6dad74E87E7f45c94f6283BF0f45)
iDAI(0x298f243aD592b6027d4717fBe9DeCda668E3c3A8)
```

Euler (not using address mapper)

### **IronBank (providerName = “IronBank”)**
```
cyWETH(0x41c84c0e2EE0b740Cf0d31F63f3B6F627DC6b393)
cyDAI(0x8e595470Ed749b85C6F7669de83EAe304C2ec68F)
cyWBTC(0x8Fc8BFD80d6A9F17Fb98A373023d72531792B431)
cyUSDC(0x76Eb2FE28b36B3ee97F3Adae0C69606eeDB2A37c)
```

### **MorphoAaveV2 (providerName = “Aave_V2”)**
```
aDAI(0x028171bCA77440897B824Ca71D1c56caC55b68A3)
aUSDC(0xBcca60bB61934080951369a648Fb03DF4F96263C)
aWBTC(0x9ff58f4fFB29fA2266Ab25e75e2A8b3503311656)
aWETH(0x030bA81f1c18d280636F32af80b9AAd02Cf0854e)
```

### **MorphoCompound (providerName = “Compound”)**
```
cDAI(0x5d3a536E4D6DbD6114cc1Ead35777bAB948E3643)
cETH(0x4Ddc2D193948926D02f9B1fE9e1daa0718270ED5)
cUSDC(0x39AA39c021dfbaE8faC545936693aC917d5E7563)
cWBTC(0xC11b1268C1A384e55C48c2391d8d480264A3A7F4)
```

### **WePiggy (providerName = “WePiggy”)**
```
pETH(0x27A94869341838D5783368a8503FdA5fbCd7987c)
pDAI(0x85166b72c87697a6acfF24101B43Fd54fE28a179)
pUSDC(0xf8E5b9738BF63ADFFf36a849F9b9C9617c8D8c1f)
pWBTC(0xc12B9D620bFCB48be3e0CCbf0ea80C717333b46F)
```

## optimism

AaveV3 (not using address mapper)

### **DForce (providerName = “DForce”)**
```
iWBTC(0x24d30216c07Df791750081c8D77C83cc8b06eB27)
iDAI(0x5bedE655e2386AbC49E2Cc8303Da6036bF78564c)
iETH(0xA7A084538DE04d808f20C785762934Dd5dA7b3B4)
iUSDC(0xB344795f0e7cf65a55cB0DDe1E866D46041A2cc2)
```

### **Hundred (providerName = “Hundred”)**
```
hETH(0xE8F12f5492Ec28609D2932519456b7436d6c93bD)
hUSDC(0x103f2CA2148B863942397dbc50a425cc4f4E9A27)
hWBTC(0x090a00A2De0EA83DEf700B5e216f87a5D4F394FE)
hDAI(0x607312a5C671D0C511998171e634DE32156e69d0)
```

### **IronBank (providerName = “IronBank”)**
```
iWETH(0x17533a1bDe957979E3977EbbFBC31E6deeb25C7d)
iUSDC(0x1d073cf59Ae0C169cbc58B6fdD518822ae89173a)
iDAI(0x049E04bEE77cFfB055f733A138a2F204D3750283)
iWBTC(0xcdb9b4db65C913aB000b40204248C8A53185D14D)
```

### **WePiggy (providerName = “WePiggy”)**
```
pETH(0x8e1e582879Cb8baC6283368e8ede458B63F499a5)
pUSDC(0x811Cd5CB4cC43F44600Cfa5eE3F37a402C82aec2)
pDAI(0xc12B9D620bFCB48be3e0CCbf0ea80C717333b46F)
pWBTC(0x48a5322c3021d5eD5CE4293112141045d12c7EFC)
```

## polygon

AaveV2 (not using address mapper)

AaveV3 (not using address mapper)

### **DForce (providerName = “DForce”)**
```
iUSDC(0x5268b3c4afb0860D365a093C184985FCFcb65234)
iDAI(0xec85F77104Ffa35a5411750d70eDFf8f1496d95b)
iWETH(0x0c92617dF0753Af1CaB2d9Cc6A56173970d81740)
iWBTC(0x94a14Ba6E59f4BE36a77041Ef5590Fe24445876A)
```

### **Hundred (providerName = “Hundred”)**
```
hDAI(0xE4e43864ea18d5E5211352a4B810383460aB7fcC)
hETH(0x243E33aa7f6787154a8E59d3C27a66db3F8818ee)
hWBTC(0xb4300e088a3AE4e624EE5C71Bc1822F68BB5f2bc)
hUSDC(0x607312a5C671D0C511998171e634DE32156e69d0)
```

### **0vix (providerName = “0vix”) - with a zero**
```
oWBTC(0x3B9128Ddd834cE06A60B0eC31CCfB11582d8ee18)
oDAI(0x2175110F2936bf630a278660E9B6E4EFa358490A)
oWETH(0xb2D9646A1394bf784E376612136B3686e74A325F)
oUSDC(0xEBb865Bf286e6eA8aBf5ac97e1b56A76530F3fBe)
```

### **WePiggy (providerName = “WePiggy”)**
```
pUSDC(0x12D803497D1e58dD4D4A4F455D754f1d0F937C8b)
pDAI(0x5cFad792C4Df1323188180778AeC58E00eAcE32a)
pWETH(0xf4B6d5d432F1C7A9EfC9E0b04acDe479F9FD1f72)
pWBTC(0xf19200b30a0416322d58e6B6b1d6B5F832936729)
```

**Comment by 0xdcota** [comment]
Update.

## Polygon
### Compound V3 (providerName = "Compound_V3")
BorrowingVault mappings
WETH => USDC => cUSDCv3(0xF25212E676D1F7F89Cd72fFEe66158f541246445)
WBTC => USDC => cUSDCv3(0xF25212E676D1F7F89Cd72fFEe66158f541246445)
MATIC => USDC => cUSDCv3(0xF25212E676D1F7F89Cd72fFEe66158f541246445)

Where:
USDC == 0x2791Bca1f2de4661ED88A30C99A7a9449Aa84174
WETH == 0x7ceB23fD6bC0adD59E62ac25578270cFf1b9f619
WBTC == 0x1BFD67037B42Cf73acF2047067bd4F2C47D9BfD6
MATIC == native key: 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE
